CMakeLists.txt

```diff
@@ -27,22 +27,45 @@ if (NOT CMAKE_INSTALL_PREFIX)
     set(CMAKE_INSTALL_PREFIX "./cmake_build")
 endif (NOT CMAKE_INSTALL_PREFIX)
 
-if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-    message(STATUS "==============OSX Environment================")
+OPTION (USE_OpenMP "Use OpenMP to enable <omp.h>" ON)
+
+# TODO: Distinguish Apple M1 chip from other Apple chips.
+IF (APPLE)
```

Comment:
to refer CMakeLists of root dir,  use MACOS as a variable.

Reply:
done.